server/src/main/java/org/elasticsearch/cluster/routing/IndexRouting.java

```diff
@@ -147,11 +149,17 @@ public void checkIndexSplitAllowed() {}
 
     private abstract static class IdAndRoutingOnly extends IndexRouting {
         private final boolean routingRequired;
+        private final IndexVersion creationVersion;
+        private final boolean isDataStream;
```

Comment:
Like I mentioned let's for now only use index mode as a heuristic to use the new uuid generator.